src/Compilers/CSharp/Portable/Symbols/Source/SourceEventSymbol.cs

```diff
@@ -484,7 +484,7 @@ protected void CheckModifiersAndType(DiagnosticBag diagnostics)
             }
             else if (IsAbstract && IsVirtual)
             {
-                diagnostics.Add(ErrorCode.ERR_AbstractNotVirtual, location, this);
+                diagnostics.Add(ErrorCode.ERR_AbstractNotVirtual, location, this.GetAttributeTarget().ToString().ToLower(), this);
```

Comment:
>this.GetAttributeTarget().ToString().ToLower() [](start = 76, length = 46)

There several issue with this code:
- it confuses  a reader because it uses API that is about attributes, but the situation we complain about has nothing to do with them.
- It looks like it gets an enum value, then calls ```ToString().ToLower()``` on it, which result in a string that is never localized.
- It prematurely does work of producing strings from raw data before anyone requested the message.  

It looks like we already have an error ERR_TypeArgsNotAllowed that needs the same information for its message, see if we can do the same thing here.